spec/system/post_index_spec.rb

```diff
@@ -0,0 +1,78 @@
+require 'rails_helper'
+
+RSpec.describe 'Post Index', type: :system do
+  before do
+    driven_by(:rack_test)
+  end
+
+  user = User.create(name: 'John Doe', photo: 'https://short.url/example', posts_counter: 5)
+
+  it "displays the user's profile picture on user posts page" do
+    visit user_posts_path(user_id: user.id)
+
+    expect(page).to have_css("img[src='#{user.photo}']")
+  end
```

Comment:
- [ ] I observed that you omitted one test (**I can see a section for pagination if there are more posts than fit on the view**), kindly implement this in accordance to the project requirement. 👍